adam-core/src/main/scala/org/bdgenomics/adam/rdd/variant/GenotypeRDD.scala

```diff
@@ -65,37 +71,108 @@ private[adam] class GenotypeArraySerializer extends IntervalArraySerializer[Refe
 object GenotypeRDD extends Serializable {
 
   /**
-   * Builds a GenotypeRDD without a partition map.
+   * An RDD containing genotypes called in a set of samples against a given
+   * reference genome.
    *
-   * @param rdd The underlying RDD.
-   * @param sequences The sequence dictionary for the RDD.
-   * @param samples The samples for the RDD.
-   * @param headerLines The header lines for the RDD.
-   * @return A new GenotypeRDD.
+   * @param rdd Called genotypes.
+   * @param sequences A dictionary describing the reference genome.
+   * @param samples The samples called.
+   * @param headerLines The VCF header lines that cover all INFO/FORMAT fields
+   *   needed to represent this RDD of Genotypes.
    */
   def apply(rdd: RDD[Genotype],
             sequences: SequenceDictionary,
             samples: Seq[Sample],
+            headerLines: Seq[VCFHeaderLine] = DefaultHeaderLines.allHeaderLines): GenotypeRDD = {
+    RDDBoundGenotypeRDD(rdd, sequences, samples, headerLines, None)
+  }
+
+  /**
+   * An RDD containing genotypes called in a set of samples against a given
+   * reference genome, populated from a SQL Dataset.
+   *
+   * @param ds Called genotypes.
+   * @param sequences A dictionary describing the reference genome.
+   * @param samples The samples called.
+   * @param headerLines The VCF header lines that cover all INFO/FORMAT fields
+   *   needed to represent this RDD of Genotypes.
+   */
+  def apply(ds: Dataset[GenotypeProduct],
+            sequences: SequenceDictionary,
+            samples: Seq[Sample],
             headerLines: Seq[VCFHeaderLine]): GenotypeRDD = {
-    GenotypeRDD(rdd, sequences, samples, headerLines, None)
+    DatasetBoundGenotypeRDD(ds, sequences, samples, headerLines)
   }
 }
 
-/**
- * An RDD containing genotypes called in a set of samples against a given
- * reference genome.
- *
- * @param rdd Called genotypes.
- * @param sequences A dictionary describing the reference genome.
- * @param samples The samples called.
- * @param headerLines The VCF header lines that cover all INFO/FORMAT fields
- *   needed to represent this RDD of Genotypes.
- */
-case class GenotypeRDD(rdd: RDD[Genotype],
-                       sequences: SequenceDictionary,
-                       @transient samples: Seq[Sample],
-                       @transient headerLines: Seq[VCFHeaderLine],
-                       optPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]]) extends MultisampleAvroGenomicRDD[Genotype, GenotypeRDD] {
+case class ParquetUnboundGenotypeRDD private[rdd] (
+    private val sc: SparkContext,
+    private val parquetFilename: String,
+    sequences: SequenceDictionary,
+    @transient samples: Seq[Sample],
+    @transient headerLines: Seq[VCFHeaderLine]) extends GenotypeRDD {
+
+  protected lazy val optPartitionMap = sc.extractPartitionMap(parquetFilename)
+
+  lazy val rdd: RDD[Genotype] = {
+    sc.loadParquet(parquetFilename)
+  }
+
+  lazy val dataset = {
```

Comment:
I also just noticed that `Coverage` is the special case here, which breaks my suggestion.